src/array_borsh.rs

```diff
@@ -0,0 +1,101 @@
+use crate::imp_prelude::*;
+use crate::IntoDimension;
+use alloc::vec::Vec;
+use borsh::{BorshDeserialize, BorshSerialize};
+use core::ops::Deref;
+
+/// **Requires crate feature `"borsh"`**
+impl<I> BorshSerialize for Dim<I>
+where
+    I: BorshSerialize,
+{
+    fn serialize<W: borsh::io::Write>(&self, writer: &mut W) -> borsh::io::Result<()> {
+        <I as BorshSerialize>::serialize(&self.ix(), writer)
+    }
+}
+
+/// **Requires crate feature `"borsh"`**
+impl<I> BorshDeserialize for Dim<I>
+where
+    I: BorshDeserialize,
+{
+    fn deserialize_reader<R: borsh::io::Read>(reader: &mut R) -> borsh::io::Result<Self> {
+        <I as BorshDeserialize>::deserialize_reader(reader).map(Dim::new)
```

Comment:
I would be worried that we have to break compatibility here if we change how dimensions are stored?